### PR TITLE
v1.2 docs omnibus

### LINKED ISSUE #1368: Update `modules` documentation
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/dbt-jinja-functions/modules

### What part(s) of the page would you like to see updated?

Add `itertools` subsection

### Additional information

Depends on: https://github.com/dbt-labs/dbt-core/pull/5140

### LINKED ISSUE #1627: File selector method
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/node-selection/methods

### What part(s) of the page would you like to see updated?

Add the `file:` selector method, which allows you to:
```
dbt --select some_model.sql
```
Notice the inclusion of `.sql`

### Additional information

https://github.com/dbt-labs/dbt-core/pull/5241

### LINKED ISSUE #1628: Selectors referencing other selectors
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/node-selection/yaml-selectors

### What part(s) of the page would you like to see updated?

YAML selectors can now reference other selectors:
```yml
selectors:
  - name: default
    definition:
      union:
        - intersection:
          - tag: foo
          - tag: bar
  - name: inherited
    definition:
      union:
        - method: selector
          value: default
        - exclude:
          - tag: bar
```
This is useful in cases where one selector includes complex node selection logic, and another selector wants only to extend/override the same selected subset in one specific way — without having to muck around with [yaml anchors](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/).

Obviously, if users create "cycles" of selectors that depend on each other, they'll encounter an error. This is definitely a feature targeting users of advanced node selection capabilities.

### Additional information

https://github.com/dbt-labs/dbt-core/pull/4827

### LINKED ISSUE #1635: Add set/zip to dbt-Jinja reference
### Contributions

- [ ] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/dbt-jinja-functions

### What part(s) of the page would you like to see updated?

- `set`
- `try_set`
- `zip`
- `try_zip`

### Additional information

https://github.com/dbt-labs/dbt-core/pull/5107

### LINKED ISSUE #1667: Manifest v6
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

- https://docs.getdbt.com/reference/artifacts/manifest-json
- https://docs.getdbt.com/guides/migration/versions/upgrading-to-v1.2

### What part(s) of the page would you like to see updated?

Top of page: `v5` &rarr; `v6` (text + link)

Similar note to the one we've had in previous migration guides:
> ### For consumers of dbt artifacts (metadata)
> The manifest schema version has updated to v6. The only change is to the default value of config for parsed nodes.

### Additional information

- https://github.com/dbt-labs/dbt-core/issues/5417
- https://github.com/dbt-labs/dbt-core/pull/5430
- https://github.com/dbt-labs/schemas.getdbt.com/pull/13

### LINKED ISSUE #1687: New global-ish configs: TARGET_PATH + LOG_PATH
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

- https://docs.getdbt.com/reference/global-configs
- https://docs.getdbt.com/reference/dbt_project.yml
- https://docs.getdbt.com/reference/project-configs/log-path
- https://docs.getdbt.com/reference/project-configs/target-path

### What part(s) of the page would you like to see updated?

dbt-core v1.2 adds support for overriding the `log-path` and `target-path` set in `dbt_project.yml`, via:
- CLI flag: `--log-path`, `--target-path`
- Env var: `DBT_LOG_PATH`, `DBT_TARGET_PATH`

For both types of global/runtime configs, the precedence order is:
1. CLI flag (`dbt run --log-path=other_logs`)
2. Env var (`DBT_LOG_PATH=other_logs dbt run`)
3. Yaml file config (`dbt_project.yml`)

In older versions of dbt, option 2 was already achievable by configuring it in the yaml file like:
```yml
# dbt_project.yml
log-path: "{{ env_var('DBT_LOG_PATH', 'logs') }}"
```
Going forward, that behavior will be implicitly true for all projects.

### Additional information

https://github.com/dbt-labs/dbt-core/pull/5402

These are "global [project] configs," subtly different from our existing "global [user/profile] configs." The difference is that `log-path` and `target-path` are traditionally defined in `dbt_project.yml` (per project), instead of the `config:` block in `profiles.yml` (per machine).

The source of that tradition is, `dbt_project.yml` is always in the root directory from which dbt is run, so runtime paths (logs, target, etc) are relative to the project file. Whereas `profiles.yml` is traditionally in a hidden folder on the machine, outside the root directory / version-controlled file system. (We are thinking about changing that: https://github.com/dbt-labs/dbt-core/issues/5411, https://github.com/dbt-labs/dbt-core/pull/5425)

**Unfortunately,** it looks like the implementation added support for these flags in a slightly different position from the standard "global config" location. I would've expected `dbt --log-path=another run`, but it's actually `dbt run --log-path=another`. Oh well 🤷 we can fix it in post, to make it work in both positions.

(One more reason that, with our work around splitting out + refactoring the `dbt-core` CLI, we should look to make position matter _not at all_, and accept flags/arguments wherever they're valid.)

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves #1368 (however, see https://github.com/dbt-labs/dbt-core/pull/5107#issuecomment-1183796139)
resolves #1667
resolves #1635
resolves #1627
resolves #1628
resolves #1687

## Description & motivation
Resolve lots of little issues, to get as much as possible live ahead of RC announcement

## Prerelease docs
If this change is related to functionality in a prerelease version of dbt (delete if not applicable):
- [x] I've added versioning components, as described in ["Versioning Docs"](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/versioningdocs.md)
- [x] I've added a note to the prerelease version's [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/zip.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+```
+{% set my_list_a = 12 %}
+{% set my_list_b = ['alice', 'bob'] %}
+{% set my_zip = zip(my_list_a, my_list_b, default = []) | list %}
+{% do log(my_zip) %}  {# [] #}
+```
+
+### try_set
```
Comment (dataders) [inline website/docs/reference/dbt-jinja-functions/zip.md]:
```suggestion
### try_zip
```

File: website/docs/reference/dbt-jinja-functions/set.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: "set"
+id: "set"
+---
+
+### set
+
+_Not to be confused with the `{% set = ... %}` expression in Jinja!_
```
Comment (dataders) [inline website/docs/reference/dbt-jinja-functions/set.md]:
```suggestion
_Not to be confused with the `{% set foo = "bar" ... %}` expression in Jinja!_
```

File: website/docs/reference/global-configs.md [hunk 3]
```diff
@@ -269,6 +269,27 @@ config:
 
 </File>
 
+<VersionBlock firstVersion="1.2">
+
+### Project paths
+
+By default, dbt will write logs to a directory named `logs/` and all other artifacts to a directory named `target/`, both as relative paths to `dbt_project.yml` of the active project—that is, the root directory from which dbt is run.
+
+Unlike other global configs, project paths are configured in `dbt_project.yml` rather than the `config:` block in `profiles.yml`:
```
Comment (dataders) [inline website/docs/reference/global-configs.md]: remind me why again?

File: website/docs/reference/node-selection/methods.md [hunk 4]
```diff
@@ -73,6 +73,19 @@ selectors unambiguous.
 
 </VersionBlock>
 
+<VersionBlock firstVersion="1.2">
+
+### The "file" method
+The `file` method can be used to select a model by its filename, including the file extension (`.sql`).
```
Comment (dataders) [inline website/docs/reference/node-selection/methods.md:79]: worth noting that if you make a model file with a different extension than `.sql` you cannot select it with this syntax? e.g.
`dbt run some_model.xlsx`

so the change is mostly just ergonomic? relatedly, before I could pass the absolute filename ending with `.sql` (inclusive of path), but I could never before just name a file with `.sql`? TIL

**Review by dataders — APPROVED** [review]
wow love that `<VersionBlock firstVersion="1.2">` convention. it'll be interesting to see the impact that many iterations of adding `VersionBlock`s has. again, I have a small suggestion and some questions/ideas. I'm so impressed you cranked this all out as fast as you did!

File: website/docs/guides/migration/versions/06-upgrading-to-v1.2.md [hunk 5]
```diff
@@ -13,14 +13,24 @@ title: "Upgrading to v1.2 (prerelease)"
...
+
+For users of [state-based selection](understanding-state): This release also includes new logic declaring forwards compatibility for older manifest versions. While running dbt Coree v1.2, it should be possible to use `state:modified --state ...` selection against a manifest produced by dbt Core v1.0 or v1.1.
+
 ## For maintainers of adapter plugins
 
 See GitHub discussion [dbt-labs/dbt-core#5468](https://github.com/dbt-labs/dbt-core/discussions/5468) for detailed information
 
 ## New and changed documentation
 
 - **[Grants](/reference/resource-configs/grants)**: You should now manage access to the datasets you're producing with dbt by using grants instead of using hooks.  If you already use post-hook to apply simple grants, moving to the grants feature will allow you to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) up your duplicated or boilerplate code.
```
Comment (runleonarun) [inline website/docs/guides/migration/versions/06-upgrading-to-v1.2.md]: @jtcohen6 should we frame this like the others "Resource configs now include [Grants](..."

File: website/docs/reference/project-configs/target-path.md [hunk 6]
```diff
@@ -17,6 +17,17 @@ Optionally specify a custom directory where compiled files (e.g. compiled models
 ## Default
 By default, dbt will write compiled files to the `target` directory, i.e. `target-path: target`
 
+<VersionBlock firstVersion="1.2">
+## Configuration
+
+The target path can also be set, in the manner of a ["global" config](global-configs), via:
+- `--target-path` CLI flag
+- `DBT_TARGET_PATH` environment variable
+
```
Comment (runleonarun) [inline website/docs/reference/project-configs/target-path.md:27]: I know we show this at the top of the page, but should we include this bullet too for completeness:
>* `target-path` dbt_project.yml

Comment (runleonarun) [reply website/docs/reference/project-configs/target-path.md:27]: If we add the dbt_project.yml bullet here, we should do the same for `log-path`

**Review by runleonarun — APPROVED** [review]
Thank you for writing this content, Jeremy!

File: website/docs/reference/dbt-jinja-functions/zip.md (hunk 1 above)
Comment (jtcohen6) [reply website/docs/reference/dbt-jinja-functions/zip.md]: Totally right. Also will need updating after we rename these to `zip_strict` + `set_strict`: https://github.com/dbt-labs/dbt-core/issues/5475

File: website/docs/reference/global-configs.md (hunk 3 above)
Comment (jtcohen6) [reply website/docs/reference/global-configs.md]: that's just how it is, buster!

jk I'll add another sentence here

File: website/docs/reference/node-selection/methods.md (hunk 4 above)
Comment (jtcohen6) [reply website/docs/reference/node-selection/methods.md:79]: > worth noting that if you make a model file with a different extension than .sql you cannot select it with this syntax?

it also would not be parsed / understood as a dbt model: [`.sql` only](https://github.com/dbt-labs/dbt-core/blob/f988f76fccc1878aaf8d8631c05be3e9104b3b9a/core/dbt/parser/read_files.py#L154) (soon `.py`, too!)

> so the change is mostly just ergonomic? relatedly, before I could pass the absolute filename ending with `.sql` (inclusive of path), but I could never before just name a file with `.sql`? TIL

exactly right!

File: website/docs/reference/project-configs/target-path.md (hunk 6 above)
Comment (jtcohen6) [reply website/docs/reference/project-configs/target-path.md:27]: good call!

File: website/docs/guides/migration/versions/06-upgrading-to-v1.2.md (hunk 5 above)
Comment (jtcohen6) [reply website/docs/guides/migration/versions/06-upgrading-to-v1.2.md]: I do feel there's a distinction — **grants** is a totally new feature, and new docs page, whereas the other bullets are describing extensions of / additions to existing pages.

I can definitely take a second pass here

File: website/docs/reference/dbt-jinja-functions/zip.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+---
+title: "zip"
+id: "zip"
+---
+
+### zip
+
+The `zip` context method can be used to used to return an iterator of tuples, where the i-th tuple contains the i-th element from each of the argument iterables. ([Python docs](https://docs.python.org/3/library/functions.html#zip))
+        :param
```
Comment (bbrewington) [inline website/docs/reference/dbt-jinja-functions/zip.md:9]: @jtcohen6 was this a typo?  or are these `:param` items meant to be there?  Glad to PR the change into the docs, just wanted to confirm first

**Pull request merged**